File: android/src/main/java/com/MythiCode/camerakit/CamerakitPlugin.java
package com.MythiCode.camerakit;

import androidx.annotation.NonNull;

import io.flutter.embedding.engine.plugins.FlutterPlugin;
import io.flutter.embedding.engine.plugins.activity.ActivityAware;
import io.flutter.embedding.engine.plugins.activity.ActivityPluginBinding;
import io.flutter.plugin.common.BinaryMessenger;
import io.flutter.plugin.platform.PlatformViewRegistry;

/**
 * CamerakitPlugin
 */
public class CamerakitPlugin implements FlutterPlugin, ActivityAware {
    private PlatformViewRegistry registry;
    private BinaryMessenger messenger;

    @Override
    public void onAttachedToEngine(@NonNull FlutterPluginBinding flutterPluginBinding) {
        registry = flutterPluginBinding.getPlatformViewRegistry();
        messenger = flutterPluginBinding.getBinaryMessenger();
    }

    @Override
    public void onDetachedFromEngine(@NonNull FlutterPluginBinding binding) {

    }

    @Override
    public void onAttachedToActivity(@NonNull ActivityPluginBinding binding) {
        if (binding.getActivity() != null) {
            registry.registerViewFactory("plugins/camera_kit", new CameraKitFactory(messenger, binding));
        }
    }

    @Override
    public void onDetachedFromActivityForConfigChanges() {

    }

    @Override
    public void onReattachedToActivityForConfigChanges(@NonNull ActivityPluginBinding binding) {

    }

    @Override
    public void onDetachedFromActivity() {

    }
}

File: android/src/main/kotlin/com/MythiCode/camerakit/CamerakitPlugin.kt
package com.MythiCode.camerakit

import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.embedding.engine.plugins.FlutterPlugin.FlutterPluginBinding
import io.flutter.embedding.engine.plugins.activity.ActivityAware
import io.flutter.embedding.engine.plugins.activity.ActivityPluginBinding
import io.flutter.plugin.common.BinaryMessenger
import io.flutter.plugin.platform.PlatformViewRegistry

/**
 * CamerakitPlugin
 */
class CamerakitPlugin : FlutterPlugin, ActivityAware {
    private var registry: PlatformViewRegistry? = null
    private var messenger: BinaryMessenger? = null
    override fun onAttachedToEngine(flutterPluginBinding: FlutterPluginBinding) {
        registry = flutterPluginBinding.platformViewRegistry
        messenger = flutterPluginBinding.binaryMessenger
    }

    override fun onDetachedFromEngine(binding: FlutterPluginBinding) {}
    override fun onAttachedToActivity(binding: ActivityPluginBinding) {
        binding.activity
        registry!!.registerViewFactory("plugins/camera_kit", CameraKitFactory(messenger!!, binding))
    }

    override fun onDetachedFromActivityForConfigChanges() {}
    override fun onReattachedToActivityForConfigChanges(binding: ActivityPluginBinding) {}
    override fun onDetachedFromActivity() {}
}